File: solutions/go/raindrops/2/raindrops.go
package raindrops

import "strconv"

type Sound struct {
	number int
	value  string
}

func Convert(number int) string {
	res := ""
	sounds := []Sound{
		{3, "Pling"},
		{5, "Plang"},
		{7, "Plong"},
	}

	for _, sound := range sounds {
		if number%sound.number == 0 {
			res += sound.value
		}
	}

	if res == "" {
		res = strconv.Itoa(number)
	}

	return res
}

File: solutions/scala/raindrops/1/src/main/scala/Raindrops.scala
object Raindrops {
  def convert(n: Int): String = {
    var res: Vector[String] = Vector.empty
    if (n % 3 == 0) res = res :+ "Pling"
    if (n % 5 == 0) res = res :+ "Plang"
    if (n % 7 == 0) res = res :+ "Plong"
    if (res.isEmpty) return n.toString
    res.mkString
  }
}

